packages/core/code-loading/transformSources.js

```diff
@@ -12,24 +12,63 @@ const modulifyTransform = (options, entry) => {
   return Object.assign({}, entry, { source })
 }
 
-const translateToJscad = (options, entry) => {
-  // console.log('translateToJscad', entry)
-  const { apiMainPath } = options
-  const deserializeStl = require('@jscad/io').stlDeSerializer.deserialize
-  const source = deserializeStl(entry.source, entry.name, options)
-  // const ext = 'js'
-  // const name = entry.name.split('.') + ext
-  // const fullPath = entry.fullPath.split('.') + ext
-  return Object.assign({}, entry, { source })
+/*
+ * Create a new entry for a script (JSCAD) from the given entry and source
+ */
+const createJscadEntry = (entry, source) => {
+  const ext = 'jscad'
+  const name = entry.name.substring(0, entry.name.lastIndexOf('.') + 1) + ext
+  const fullPath = '/' + name
+
+  return Object.assign({}, entry, { ext, name, fullPath, source })
+}
+
+const transformAmfToJscad = (options, entry) => {
```

Comment:
these are all great !
With a big catch, please read my comment in the overral review